mountpoint-s3/src/prefetch.rs

```diff
@@ -139,6 +148,12 @@ where
             "read"
         );
 
+        // Currently, preferred part size is the current read size.
+        // This is a naive assumption that the read size will be the same for most sequential
+        // read and it can be aligned to the size of prefetched data. Even so, it's worth to
+        // try since the performance will be a lot better when the conditions are met.
+        self.preferred_part_size = length;
```

Comment:
We probably want to put some bounds on this value (e.g. if someone does a 1 byte read, that's probably not the part size we want).